ouroboros-consensus/src/Ouroboros/Consensus/Mempool/API.hs

```diff
@@ -100,10 +100,28 @@ data Mempool m blk = Mempool {
       -- index of transaction hashes that have been included on the blockchain.)
       addTxs :: [GenTx blk] -> m [(GenTx blk, ApplyTxErr blk)]
 
-      -- | Get all transactions in the mempool (oldest to newest)
+      -- |
+    , syncState :: STM m ()
+
+      -- | Get all transactions in the mempool along with their associated
+      -- ticket numbers (oldest to newest).
       --
-      -- Guarantees that these transactions will be valid (when applied strictly
-      -- in order) with respect to a call to 'getLedgerState' run /in the same
-      -- transaction/.
-    , getTxs :: STM m (Seq (GenTx blk))
+      -- n.b. This function provides /no guarantee/ that the resulting
+      -- transactions will be valid. However, it is guaranteed that these
+      -- transactions will be valid if this function is called immediately
+      -- following a call to 'syncState', /within the same transaction/.
+    , getTxs :: STM m (Seq (GenTx blk, idx))
+
+      -- | Get all transactions in the mempool, along with their associated
+      -- ticket numbers, which are associated with a ticket number greater
+      -- than the one provided.
+    , getTxsAfter :: idx -> STM m (Seq (GenTx blk, idx))
+
+      -- | Get a specific transaction from the mempool by its ticket number,
+      -- if it exists.
+    , getTx :: idx -> STM m (Maybe (GenTx blk))
+
+      -- | Represents the initial value at which the transaction ticket number
+      -- counter will start (i.e. the zeroth ticket number).
+    , zeroIdx :: idx
```

Comment:
What's the reason for making this part of the API? (I'm just wondering)

**EDIT:** because of `implGetTxs = (flip implGetTxsAfter) zeroTicketNo`

Reply:
Alternatively, we could impose `Enum idx` and `Bounded idx`.

Reply:
I asked for it (or equiv) because it means I can use `getTxsAfter` in all cases (by starting with `zeroIdx`), rather than have to use a combo of `getTxs` & `getTxsAfter`. This simplifies the tx submission protocol handler.